plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -280,7 +335,34 @@ _transaction_progress_changed_cb (FlatpakTransactionProgress *progress,
 			   gs_app_get_progress (app), percent);
 		return;
 	}
+
 	gs_app_set_progress (app, percent);
+
+	// TODO: Improve heuristics to get progress for proxy app
+	proxy_app = g_hash_table_lookup (self->apps_proxies, app);
+	if (proxy_app) {
+		GsAppList *proxied_apps;
+		guint64 total_expected_size_installed = 0;
+		guint64 total_size_installed = 0;
+
+		proxied_apps = gs_app_get_related (proxy_app);
+		for (guint i = 0; i < gs_app_list_length (proxied_apps); ++i) {
+			GsApp *proxied_app = gs_app_list_index (proxied_apps, i);
+			guint64 proxied_app_size_installed;
+			guint proxied_app_progress;
+
+			proxied_app_size_installed = gs_app_get_size_installed (proxied_app);
```

Comment:
would be good to check for if proxied_app_size_installed is 0 or GS_APP_SIZE_UNKNOWABLE just in case

Reply:
You mean check and warn if 0? It wouldn't make any diff to the logic given this variable is only used on addition/multiplication. It could give a very slight performance improvement as we could skip some computations but then again we would gain an extra condition check in every call on this loop if the size is not 0.

I am already protecting against `total_expected_size_installed` when computing the progress for the proxy app.

Reply:
Ok, I see now GS_APP_SIZE_UNKNOWABLE is not 0. I will update.

Reply:
Forced pushed with this update, didn't add any warning though.

Also checked and many other places are invoking `gs_app_get_size_installed (app)` without considering `GS_APP_SIZE_UNKNOWABLE` which in reality is only returned if `GS_IS_APP(app)` fails.